### PR TITLE
improve test coverage

### PR DESCRIPTION
**Pull request by ChunkyProgrammer**
i expect some tests to be failing at the moment

### CODE REVIEW

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Comment by PikachuEXE** [comment]
If they are failing now how to do we tell if other PRs can be merged or not after this PR is merged?

**Comment by PikachuEXE** [comment]
If they are failing now how to do we tell if other PRs can be merged or not after this PR is merged?

**Review by PrestonN — APPROVED** [review]
LGTM

**Pull request merged**
